gensim/similarities/nmslib.py

```diff
@@ -0,0 +1,225 @@
+            If the model is None, index and labels are not initialized.
+            In that case please load or init the index and labels by yourself.
+        index_params : dict, optional
+            index_params for Nmslib indexer.
+        query_time_params : dict, optional
+            query_time_params for Nmslib indexer.
+
+        """
+        if index_params is None:
+            index_params = {'M': 100, 'indexThreadQty': 1, 'efConstruction': 100, 'post': 0}
+        if query_time_params is None:
+            query_time_params = {'efSearch': 100}
+
+        self.index = None
+        self.labels = None
+        self.model = model
+        self.index_params = index_params
+        self.query_time_params = query_time_params
+
+        if model:
+            if isinstance(self.model, Doc2Vec):
+                self._build_from_doc2vec()
+            elif isinstance(self.model, (Word2Vec, FastText)):
+                self._build_from_word2vec()
+            elif isinstance(self.model, (WordEmbeddingsKeyedVectors, KeyedVectors)):
+                self._build_from_keyedvectors()
+            else:
+                raise ValueError("model must be a Word2Vec, Doc2Vec, FastText or KeyedVectors instance")
+
+    def save(self, fname, protocol=2):
+        """Save this NmslibIndexer instance to a file.
+
+        Parameters
+        ----------
+        fname : str
+            Path to the output file,
+            will produce 2 files: `fname` - parameters and `fname`.d - :class:`~nmslib.NmslibIndex`.
+        protocol : int, optional
+            Protocol for pickle.
+
+        Notes
+        -----
+        This method saves **only** the index (**the model isn't preserved**).
+
+        """
+        fname_dict = fname + '.d'
+        self.index.saveIndex(fname)
+        d = {'index_params': self.index_params, 'query_time_params': self.query_time_params, 'labels': self.labels}
+        with open(fname_dict, 'wb') as fout:
+            _pickle.dump(d, fout, protocol=protocol)
+
+    @classmethod
+    def load(cls, fname):
+        """Load a NmslibIndexer instance from a file
+
+        Parameters
+        ----------
+        fname : str
+            Path to dump with NmslibIndexer.
+
+        """
+        fname_dict = fname + '.d'
+        with open(fname_dict, 'rb') as f:
+            d = _pickle.load(f)
+        index_params = d['index_params']
+        query_time_params = d['query_time_params']
+        nmslib_instance = cls(index_params=index_params, query_time_params=query_time_params)
+        index = nmslib.init()
+        index.loadIndex(fname)
+        nmslib_instance.index = index
+        nmslib_instance.labels = d['labels']
+        return nmslib_instance
+
+    def _build_from_word2vec(self):
+        """Build an Nmslib index using word vectors from a Word2Vec model."""
+
+        self.model.init_sims()
+        return self._build_from_model(self.model.wv.vectors_norm, self.model.wv.index2word)
```

Comment:
It doesn't look like you're using this return value anywhere (same for the other build functions). Furthermore, it looks like this value will always be None.

I think it'd be clearer if you didn't return anything.

Reply:
I see. I will delete "return".

Reply:
Deleted "return".